server/src/transport/ws.rs

```diff
@@ -448,6 +440,14 @@ where
 			let rpc_service = rpc_middleware.service(rpc_service);
 
 			let fut = async move {
+				let upgraded = match hyper::upgrade::on(req).await {
```

Comment:
Yup ok, I have no idea why moving this up would break anything offhand!

Reply:
I forgot about https://github.com/hyperium/hyper/blob/master/examples/upgrades.rs#L56-#L72 so then it adds up :)